server/datastore/cached_mysql/cached_mysql.go

```diff
@@ -313,10 +317,55 @@ func (ds *cachedMysql) DeleteTeam(ctx context.Context, teamID uint) error {
 	agentOptionsKey := fmt.Sprintf(teamAgentOptionsKey, teamID)
 	featuresKey := fmt.Sprintf(teamFeaturesKey, teamID)
 	mdmConfigKey := fmt.Sprintf(teamMDMConfigKey, teamID)
+	teamKey := fmt.Sprintf(teamByIdKey, teamID)
 
 	ds.c.Delete(agentOptionsKey)
 	ds.c.Delete(featuresKey)
 	ds.c.Delete(mdmConfigKey)
+	ds.c.Delete(teamKey)
 
 	return nil
 }
+
+func (ds *cachedMysql) Team(ctx context.Context, teamID uint) (*fleet.Team, error) {
+	key := fmt.Sprintf(teamByIdKey, teamID)
+	if x, found := ds.c.Get(key); found {
+		if team, ok := x.(*fleet.Team); ok {
+			return team, nil
+		}
+	}
+
+	team, err := ds.Datastore.Team(ctx, teamID)
+	if err != nil {
+		return nil, err
+	}
+	ds.c.Set(key, team, ds.scheduledQueriesExp)
+	return team, nil
+}
+
+func (ds *cachedMysql) ListQueries(ctx context.Context, opt fleet.ListQueryOptions) ([]*fleet.Query, error) {
```

Comment:
I just realized this method is used by the user and by osquery agents.

We definitely don't want caching on the "user side", only on the osquery side.

I can see a few options:
1. Duplicating this method for osquery (`ListQueriesLite`, lite == without stats or stuff not needed by agents) and cache that one.
2. Add the caching of ListQueries to the osquery's datastore only.

Let me know if it makes sense.

Reply:
I'll go with option 1, should help keep the query simpler.

Reply:
https://github.com/fleetdm/fleet/pull/12815/commits/557f698ea2ab4660de452eea5872c88cad60f660